server/db/driver/pg/orders.go

```diff
@@ -735,6 +735,51 @@ func completedUserOrders(ctx context.Context, dbe *sql.DB, tableName string, aid
 	return
 }
 
+// CountUserPreimageMisses retrieves the number of times the user has failed to
+// respond to a preimage request for an order
+func (a *Archiver) CountUserPreimageMisses(user account.AccountID, lastNOrders int32) (uint32, error) {
+	var count uint32
+	queryCount := func(stmt string) error {
+		ctx, cancel := context.WithTimeout(a.ctx, a.queryTimeout)
+		defer cancel()
+		var mCount uint32
+		var err error
+		if lastNOrders == -1 {
+			err = a.db.QueryRowContext(ctx, stmt, user).Scan(&mCount)
```

Comment:
`-1 = all` feature is currently unused. Are you anticipating a need?

Reply:
As with `UserMatchSummary` I'm intending to keep this for administrative purposes.  If that doesn't happen, we can remove.